queue_job_batch/models/queue_job_batch.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models, fields, api
+from odoo.addons.queue_job.job import job
+
+
+class QueueJobBatch(models.Model):
+    _name = 'queue.job.batch'
+    _inherit = ['mail.thread', 'mail.activity.mixin']
+    _description = 'Batch of jobs'
+    _log_access = False
+
+    name = fields.Char(
+        required=True,
+        readonly=True,
+        track_visibility='onchange',
+    )
+    job_ids = fields.One2many(
+        'queue.job',
+        inverse_name='job_batch_id',
+        readonly=True,
+    )
+    job_count = fields.Integer(
+        compute='_compute_job_count',
+    )
+    user_id = fields.Many2one(
+        'res.users',
+        required=True,
+        readonly=True,
+        track_visibility='onchange',
+    )
+    state = fields.Selection(
+        [
+            ('draft', 'Draft'),
+            ('progress', 'In Progress'),
+            ('finished', 'Finished')
+        ],
+        default='draft',
+        required=True,
+        readonly=True,
+        track_visibility='onchange',
+    )
+    finished_job_count = fields.Float(
+        compute='_compute_job_count',
+    )
+    failed_job_count = fields.Float(
+        compute='_compute_job_count',
+    )
+    company_id = fields.Many2one(
+        'res.company',
+        readonly=True,
+    )
+    is_read = fields.Boolean(
+        default=True
+    )
+    completeness = fields.Float(
+        compute='_compute_job_count',
+    )
+    failed_percentage = fields.Float(
+        compute='_compute_job_count',
+    )
+
+    @job
+    def check_state(self):
+        self.ensure_one()
+        if self.state == 'draft' and any(
+            job.state not in ['pending', 'enqueued'] for job in self.job_ids
+        ):
+            self.write({'state': 'progress'})
+        if self.state != 'progress':
+            return True
+        if all(job.state == 'done' for job in self.job_ids):
+            self.write({
+                'state': 'finished',
+                'is_read': False,
+            })
+        return True
+
+    @api.multi
+    def set_read(self):
+        return self.write({'is_read': True})
+
+    @api.model
+    def get_new_batch(self, name, **kwargs):
+        vals = kwargs.copy()
+        if 'company_id' in self.env.context:
+            company_id = self.env.context['company_id']
+        else:
+            company_model = self.env['res.company']
+            company_model = company_model.sudo(self.user_id)
```

Comment:
self.user_id is suspicious in a model method.
Why is this special processing to get company_id necessary, and why is this get_new_batch necessary over a regular create?

Reply:
I took the same logic used on `queue.job`:
https://github.com/OCA/queue/blob/11.0/queue_job/job.py#L445
Which should be the right approach?

Reply:
> Which should be the right approach?

I don't know. What I'm quite sure is that `self.user_id` on a model method is not going to work.
@guewen any idea?

Reply:
Maybe we can use `self.env.uid`

Reply:
> Maybe we can use self.env.uid

Yes that would be better, certainly. I still don't understand the goal of all this complex logic with sudo etc. I guess a simple create with default values for `user_id` and `company_id` would do the job, but I may be missing something.

Reply:
Reviewing the original code, it has to be replaced by `self.env.uid`. I took the code, but there was something before that was necessary
https://github.com/OCA/queue/blob/11.0/queue_job/job.py#L444
Change applied.